go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/flanksource/commons
 
-go 1.17
+go 1.20
```

Comment:
updated go version since this PR makes use of Go generics

Reply:
Also update the go version in all the `.github/workflow` files for lint etc.

Reply:
I see that this doesn't have a lint check, can you copy the one from other repos here

Reply:
sure